File: frontend/src/stores/conversation.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export interface ChatMessage {
  id: string
  conversation_id: string
  role: 'user' | 'assistant'
  content_type: 'text' | 'config' | 'result'
  payload: Record<string, unknown>
  created_at: string
}

export interface Conversation {
  id: string
  title: string
  dataset_name: string | null
  created_at: string
  updated_at: string
  messages: ChatMessage[]
}

export const useConversationStore = defineStore('conversation', () => {
  const currentId = ref<string | null>(null)
  const title = ref('')
  const datasetName = ref<string | null>(null)
  const messages = ref<ChatMessage[]>([])
  const loading = ref(false)

  function reset() {
    currentId.value = null
    title.value = ''
    datasetName.value = null
    messages.value = []
    loading.value = false
  }

  function setConversation(conv: Conversation) {
    currentId.value = conv.id
    title.value = conv.title
    datasetName.value = conv.dataset_name
    messages.value = conv.messages
  }

  function addMessage(msg: ChatMessage) {
    messages.value.push(msg)
  }

  return {
    currentId, title, datasetName, messages, loading,
    reset, setConversation, addMessage,
  }
})

File: backend/models/conversation.py
from __future__ import annotations

import uuid
from dataclasses import dataclass, field
from datetime import datetime, timezone


@dataclass
class Conversation:
    user_id: str
    title: str = "New analysis"
    dataset_name: str | None = None
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    created_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))


@dataclass
class Message:
    conversation_id: str
    role: str  # 'user' | 'assistant'
    payload: str = "{}"
    content_type: str = "text"  # 'text' | 'config' | 'result'
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    created_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
